internal/processmon/processmon.go

```diff
@@ -24,11 +22,8 @@ import (
 )
 
 var (
-	// GlobalCommandArgs are command args received while invoking this command
-	GlobalCommandArgs map[string]interface{}
-	launcher          *ProcessMon
-	netnspath         string
-	childExitStatus   = make(chan ExitStatus, 100)
+	// launcher supports only a global processMon instance
+	launcher *processMon
```

Comment:
if that's the case calling `newProcesMon` or whatever should panic if `launcher` is already set. If you can swap it, then a lock is needed.

Reply:
So swap is only possible inside a test because they cant actually write to /var/run/ whatever. newProcessMon is a private function and cant be called by anyone other than init().